tests/std/tests/P0896R4_ranges_alg_copy_n/test.cpp

```diff
@@ -9,34 +9,30 @@
 
 #include <range_algorithm_support.hpp>
 
-constexpr void smoke_test() {
-    using ranges::copy_n, ranges::copy_n_result, ranges::iterator_t;
-    using std::same_as;
+using ranges::copy_n, ranges::copy_n_result, ranges::iterator_t;
+using std::same_as;

```

Comment:
Modern test style uses `using namespace std;` to be consistent with the majority of the test base.

Reply:
So this is interesting. I thought this was to avoid collisions with the classical algorithms. As far as I can see we would need to qualify all iterator + sentinel variants that do not use a projection. Which in turn would essentially mean we would need to qualify all of them to be consistent. 

Given that I would rather not have `using namespace std;` in ranges tests. Thoughts?

`const` west ehhhh

Reply:
There's no firm policy in place yet, but we're leaning toward making the test codebase more consistent. Traditionally tests have been islands maintained by "feature owners" in whatever style they prefer (this was enormously impactful years ago when the product code style was a Lovecraftian nightmare) but we're concerned that the existence of distinct islands makes it hard to move between them. This is a barrier to newer contributors that we don't want.

Consequently, `using namespace std;` (although personally I hate it, it's de facto).

That means qualification at namespace scope, although it's easily overridden by e.g. `using ranges::copy` at namespace scope.

Reply:
Whenever I write a stackoverflow answer there is a 90% chance that I advocate against it. but I mean this is the STL so...